libs/sdk-core/src/persist/sync.rs

```diff
@@ -163,6 +169,18 @@ impl SqliteStorage {
             [],
         )?;
 
+        // sync remote swap_refunds table
+        tx.execute(
+            "
+        INSERT INTO sync.open_channel_payment_info
+         SELECT
+          payment_hash,
+          payer_amount_msat
+         FROM remote_sync.open_channel_payment_info
+         WHERE payment_hash NOT IN (SELECT payment_hash FROM sync.open_channel_payment_info);",
```

Comment:
These records should probably not be correlated by the payment hash, but by some kind of internally created checking_id. Amount + payment hash is already better. [source](https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-June/003983.html#:~:text=The%20mitigation%20is%20quite%20simple.%20Backends%20should%20either%20use%20self%2Dgenerated%20unique%20%22checking%20id%27s%22%20for%20looking%20up%20internal%20payments%20or%20use%20additional%20checks%20to%20make%20sure%20that%20the%20invoice%20details%20have%20not%20been%20messed%20around%20with%20(e.g.%2C%20asserting%20amount(A)%20%3D%3D%20amount(B)).)

Reply:
This is internal associated by the user and not involving any third party or  assumptions like were mentioned in this case.
The issue there, as I understand it, that they associated a payment hash with an internal payment (god knows what that is but certainly not LN payment) which they use to credit the account.
In our case it is only to associate an already existing invoice (with a unique payment hash) to its already deduced payer amount for fee calculations. Do you see any concrete problematic use case here?

Reply:
Yeah that's a specific case. But I think in general the payment hash doesn't identify a payment, and we should be careful not to identify the payment with a payment hash in our software. 

I don't see any problematic use case here now, but I can see how it could develop into a problematic case if software is written on top of this.

Reply:
I get your point @JssDWt but in our case it is not used to identify a payment. It is used to associated data to an invoice.
Invoices are indexed by a unique id in the sdk storage (which seems OK isn't it?) therefore I see it as a natural step to associate this data using the unique id in this specific case.

Reply:
And the unique id is the payment hash you mean? 
I don't see a problem now, but we have to be careful to never do stuff like "ah, I have an invoice here with payment hash x, so that must be the one from the unique index in my database".